### ISSUE TITLE
Avoid loading scripts multiple times

### ISSUE THREAD

**Issue by 9inpachi** [muratcorlu/ngx-script-loader#26]
Hi,

Since this is appending the scripts as HTML elements, it would make sense to check if the script tag has already been added/loaded. We could also check if all the attributes indeed exist.
I had a use case where this was quite problematic.
Let me know if this is welcomed. I will prepare a PR.

Thanks. :)

**Comment by muratcorlu** [comment]
Hi,

As I mentioned in [Readme](https://github.com/muratcorlu/ngx-script-loader#running-a-script-multiple-times) `loadScript` method indeed inject same script once to the document.

> By default `loadScript()` method loads a script only once in a session. If you want to inject same script multiple times, you can use `runScript()` method instead.

Do you think this is not enough or not working properly?

**Comment by 9inpachi** [comment]
> As I mentioned in Readme loadScript method indeed inject same script once to the document.

So I did a little more testing and it seems to work as documented. It was a problem with my code. I was dynamically creating the `ScriptService` and instantiating it multiple times. So from what I understand from the usage (didn't look at the code for this), it seems a service instance maintains which scripts are already loaded.

Do you think a use case like this (apart from my bad code) is viable where multiple `ScriptService`s might be required? If so, it might be worth checking in the DOM for loaded scripts.

And apart from all this - thanks for the extension. It's really helpful. :)

**Comment by 9inpachi** [comment]
Here's something for a reference code.

**script-loader.ts**
```ts
import { ScriptService } from "ngx-script-loader";

export async function loadSomeScript(onLoad) {
    const scriptService = new ScriptService(document); // New ScriptService for every call - bad code
    for (let i = 0; i < 10; i++) {
        await scriptService.loadScript('assets/someScript.js').toPromise();
    }
    onLoad();
}
```

**using-loader.ts**
```ts
(async () => {
  await loadSomeScript(() => {
    console.log('Loaded');
  });

  await loadSomeScript(() => {
    console.log('Loaded 2');
  });
})();
```

**Comment by muratcorlu** [comment]
I don't know if there is a use case to have multiple instances of `ScriptService`. I also need to think about side effects of checking existence of script in DOM already. I'm not completely clear about that approach for now. Current "in-memory" approach is fast, easy to manage and very flexible(I just used `shareReplay` of rxjs).

**Comment by 9inpachi** [comment]
I see. Let me know if you need any help.
And if you feel like it, you can close the issue. Thanks. :)

**Comment by muratcorlu** [comment]
Thank you. I'll create a separate issue if I'll need to work on this topic. Closing this now.
